### PR TITLE
rye: 0.6.0 -> 0.11.0

### PR DESCRIPTION
**Pull request by GaetanLepage**
###### Description of changes

Update `rye` to its latest version.
CHANGELOG: https://github.com/mitsuhiko/rye/releases/tag/0.11.0

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 244122` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>rye</li>
  </ul>
</details>

File: pkgs/development/tools/rye/default.nix [hunk 1]
```diff
@@ -10,24 +10,27 @@
...
-  version = "0.6.0";
+  version = "0.10.0";
 
   src = fetchFromGitHub {
     owner = "mitsuhiko";
     repo = pname;
-    rev = version;
+    rev = "refs/tags/${version}";
     # One of the tests runs git command so we need .git directory there
     leaveDotGit = true;
```
Comment (wegank) [inline pkgs/development/tools/rye/default.nix:20]: Could we just skip the git-dependent test?

Comment (GaetanLepage) [reply pkgs/development/tools/rye/default.nix]: It seems that all the tests now work. I removed the `leaveDotGit = true;` line and did not have to skip any test !

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/development/tools/rye/default.nix (hunk 1 above)
Comment (GaetanLepage) [reply pkgs/development/tools/rye/default.nix]: > It seems that all the tests now work. I removed the leaveDotGit = true; line and did not have to skip any test !

Well no. While updating to 0.11.0, I was needed to skip the test.

**Comment by GaetanLepage** [comment]
The darwin CI is failing:
```
ld: symbol(s) not found for architecture x86_64
```
I don't know really why it happens.

File: pkgs/development/tools/rye/default.nix [hunk 2]
```diff
@@ -35,12 +34,13 @@ rustPlatform.buildRustPackage rec {
...
   buildInputs = [
-    git
     openssl
   ]
   ++ lib.optional stdenv.isDarwin SystemConfiguration;
 
-  nativeCheckInputs = [ git ];
+  cargoTestFlags = [
+    "-- --skip=utils::test_is_inside_git_work_tree"
+  ];
```
Comment (wegank) [inline pkgs/development/tools/rye/default.nix:43]:
```suggestion
  checkFlags = [
    "--skip=utils::test_is_inside_git_work_tree"
  ];
```

File: pkgs/development/tools/rye/default.nix [hunk 3]
```diff
@@ -35,12 +34,13 @@ rustPlatform.buildRustPackage rec {
   nativeBuildInputs = [ pkg-config ];
 
   buildInputs = [
-    git
     openssl
   ]
   ++ lib.optional stdenv.isDarwin SystemConfiguration;
```
Comment (wegank) [inline pkgs/development/tools/rye/default.nix:39]:
```suggestion
  ++ lib.optionals stdenv.isDarwin [
    Libsystem
    SystemConfiguration
  ];
```

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -19651,7 +19651,7 @@ with pkgs;
   rufo = callPackage ../development/tools/rufo { };
 
   rye = callPackage ../development/tools/rye {
```
[the remark this answers is not in the document]
Comment (GaetanLepage) [reply pkgs/top-level/all-packages.nix:19653]: I changed it. Is it what you had in mind ?

[the remark this answers is not in the document]
Comment (wegank) [reply pkgs/top-level/all-packages.nix:19653]: Yes, it should now work on both darwin platforms.

File: pkgs/development/tools/rye/default.nix (hunk 3 above)
Comment (GaetanLepage) [reply pkgs/development/tools/rye/default.nix]: I am not really fluent with the darwin-specific changes. Are you satisfied with the current state ?

Comment (wegank) [reply pkgs/development/tools/rye/default.nix]: Yes, I'm just waiting for ofborg's eval.

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 244122` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>rye</li>
  </ul>
</details>

**Review by wegank — APPROVED** [review]

**Pull request merged**